cybox/objects/pdf_file_object.py

```diff
@@ -1,12 +1,43 @@
-# Copyright (c) 2015, The MITRE Corporation. All rights reserved.
+# Copyright (c) 2017, The MITRE Corporation. All rights reserved.
 # See LICENSE.txt for complete terms.
 
 from mixbox import entities
 from mixbox import fields
 
 import cybox.bindings.pdf_file_object as pdf_file_binding
 from cybox.objects.file_object import File
-from cybox.common import DateTime, PositiveInteger, String, NonNegativeInteger, Double
+from cybox.common import (DateTime, HashList, PositiveInteger, String,
+                          NonNegativeInteger, Double, HexBinary, Integer)
+
+
+def validate_pdf_object_type(instance, value):
+    if value is None:
+        return
+    elif value in PDFIndirectObject.TYPES:
+        return
+    else:
+        err = "Type must be one of %s. Received '%s'." % (PDFIndirectObject.TYPES, value)
+        raise ValueError(err)
+
+
+def validate_pdf_ref_entry_type(instance, value):
+    if value is None:
+        return
+    elif value in PDFXrefEntry.TYPES:
+        return
+    else:
+        err = "Type must be one of %s. Received '%s'." % (PDFXrefEntry.TYPES, value)
+        raise ValueError(err)
+
+
+def validate_id_string(instance, value):
+    if value is None:
+        return
+    elif len(instance) > 2:
```

Comment:
I'm pretty sure `instance` will be an instance of `PDFFileID`, which would be fine if this was an EntityList, but we might want to use `len(instance.id_string)` instead. Also, I'm thinking this should be `>=2`, since we don't want to add another one if there are already 2.